File: 5.implementation/redsun-service/src/main/java/com/redsun/service/service/OverallProgressesService.java
package com.redsun.service.service;

import com.redsun.service.entities.OverallProgresses;
import com.redsun.service.entities.Result;

/**
 * OverallProgresses Service interface
 */
public interface OverallProgressesService {
	
    // Insert.
	Result insert(final OverallProgresses overallProgresses) ;

    // Update.
	Result update(final OverallProgresses overallProgresses) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listOverallProgressesForPageAndFilter(final int itemsPerPage, final int pageNo, final OverallProgresses overallProgresses);

}

File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/OverallProgressesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('overallProgresseService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(overallProgresse) {
			var urlServer = $rootScope.makePostURL("/overallprogresse/insert");
			return $http.post(urlServer, overallProgresse);
		}
		
		// Update.
		, update: function(overallProgresseId, overallProgresse) {
			var urlServer = $rootScope.makePostURL("/overallprogresse/update/" + overallProgresseId);
			return $http.put(urlServer, overallProgresse);
		}
		
		// Delete.
		, delete: function(overallProgresseId) {
			var urlServer = $rootScope.makePostURL("/overallprogresse/delete/" + overallProgresseId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(overallProgresseId) {
			var url = $rootScope.makeGetURL("/overallprogresse/getbyid/" + overallProgresseId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, overallProgresse) {
			var url = $rootScope.makeGetURL("/overallprogresse/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: overallProgresse });
		}
		
	}// return.
});